packages/action-menu/stories/action-menu.stories.ts

```diff
@@ -116,3 +117,52 @@ export const submenu = (): TemplateResult => {
         </sp-action-menu>
     `;
 };
+
+export const controlled = (): TemplateResult => {
+    const state = {
+        snap: true,
+        grid: false,
+        guides: true,
+    };
+    function toggle(prop: keyof typeof state) {
+        return (event: Event): void => {
+            const item = event.target as MenuItem;
+            state[prop] = !state[prop];
+            // in Lit-based usage, this would be handled via render():
+            // <sp-menu-item ?selected=${this.isSomethingSelected}>
+            item.selected = state[prop];
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            document.getElementById(
+                'state-json'
+            )!.textContent = `application state: ${JSON.stringify(state)}`;
+        };
+    }
+    return html`
+        <sp-action-menu label="View">
```

Comment:
Is it possible to toggle from the `change` event instead of from each Menu Item?

Reply:
It's possible, but I think for this story we should keep it as per-item events. The point of the story is to show that a consumer can lift all control out of the component, both for listening to inputs and for setting selected state.

Not a strong opinion though, so I'm happy to migrate to `change` if you still think we should.

Reply:
Maybe it needs both then. I think of it as a `<select>` which prevent you from setting `click` events on its `<option>` elements, but I know a lot of teams already consume this element a la this demo, which I think is a less optimal consumption. Ensuring success for everyone in doing things _their_ way is that name of the game though, I guess.

Reply:
Cool, I've added state tracking to the last-changed item as well.